addons/mail/models/ir_attachment.py

```diff
@@ -71,5 +82,98 @@ def _attachment_format(self, commands=False):
                     'res_id': attachment.res_id,
                     'res_model': attachment.res_model,
                 })
-            res_list.append(res)
-        return res_list
+            attachments.append(res)
+        return attachments
+
+    @api.model
+    def _get_data_from_url(self, url):
+        """
+        This will create attachment data based on what we can read from the URL.
+        If the URL is an HTML page, this will parse the OpenGraph meta data.
+        If the URL is an image, this will create an image attachment.
+        """
+        try:
+            page = requests.get(url, timeout=1)
+        except requests.exceptions.RequestException:
+            return False
+
+        if page.status_code != requests.codes.ok:
+            return False
+
+        image_mimetype = [
+            'image/bmp',
+            'image/gif',
+            'image/jpeg',
+            'image/png',
+            'image/tiff',
+            'image/x-icon',
+        ]
+        if page.headers['Content-Type'] in image_mimetype:
+            return self._get_data_from_url_image(url)
+        elif 'text/html' in page.headers['Content-Type']:
```

Comment:
The `elif` is not needed thanks to the early return.
```suggestion
        if 'text/html' in page.headers['Content-Type']:
```